platform/ariane/platform.c

```diff
@@ -0,0 +1,287 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright (C) 2019 FORTH-ICS/CARV
+ *				Panagiotis Peristerakis <perister@ics.forth.gr>
+ */
+
+#include <sbi/riscv_encoding.h>
+#include <sbi/sbi_const.h>
+#include <sbi/sbi_platform.h>
+#include <plat/irqchip/plic.h>
+#include <plat/serial/uart8250.h>
+#include <plat/sys/clint.h>
+#include <sbi/sbi_console.h>
+#include <sbi/sbi_hart.h>
+#include <libfdt.h>
+#include <fdt.h>
+#include <plat/irqchip/plic.h>
+#include <sbi/riscv_io.h>
+
+#define ARIANE_UART_ADDR			0x10000000
+#define ARIANE_UART_FREQ			50000000
+#define ARIANE_UART_BAUDRATE			115200
+#define ARIANE_UART_REG_SHIFT			2
+#define ARIANE_UART_REG_WIDTH			4
+#define ARIANE_PLIC_ADDR			0xc000000
+#define ARIANE_PLIC_NUM_SOURCES			3
+#define ARIANE_HART_COUNT			1
+#define ARIANE_CLINT_ADDR 0x2000000
+#define PLIC_ENABLE_BASE		0x2000
+#define PLIC_ENABLE_STRIDE		0x80
+#define PLIC_CONTEXT_BASE		0x200000
+#define PLIC_CONTEXT_STRIDE		0x1000
+
+#define SBI_ARIANE_FEATURES	\
+	(SBI_PLATFORM_HAS_TIMER_VALUE | \
+	 SBI_PLATFORM_HAS_SCOUNTEREN | \
+	 SBI_PLATFORM_HAS_MCOUNTEREN | \
+	 SBI_PLATFORM_HAS_MFAULTS_DELEGATION)
+
+static volatile void *plic_base;
+
+/*
+ * Ariane platform early initialization.
+ */
+static int ariane_early_init(bool cold_boot)
+{
+	/* For now nothing to do. */
+	return 0;
+}
+
+/*
+ * Ariane platform final initialization.
+ */
+static int ariane_final_init(bool cold_boot)
+{
+void *fdt;
+
+	if (!cold_boot)
+		return 0;
+	fdt = sbi_scratch_thishart_arg1_ptr();
+	plic_fdt_fixup(fdt, "riscv,plic0");
+	return 0;
+}
+
+/*
+ * Initialize the ariane console.
+ */
+static int ariane_console_init(void)
+{
+	return uart8250_init(ARIANE_UART_ADDR,
+						 ARIANE_UART_FREQ,
+						 ARIANE_UART_BAUDRATE,
+						ARIANE_UART_REG_SHIFT,
+						ARIANE_UART_REG_WIDTH);
+}
+
+/*
+ * Write a character to the ariane console output.
+ */
+static void ariane_console_putc(char ch)
+{
+
+	uart8250_putc(ch);
+}
+
+/*
+ * Read a character from the ariane console input.
+ */
+static int ariane_console_getc(void)
+{
+	return uart8250_getc();
+}
+
+static void plic_set_thresh(u32 cntxid, u32 val)
+{
+	volatile void *plic_thresh =
+		plic_base + PLIC_CONTEXT_BASE + PLIC_CONTEXT_STRIDE * cntxid;
+	writel(val, plic_thresh);
+}
+
+static void plic_set_ie(u32 cntxid, u32 word_index, u32 val)
```

Comment:
Rather than re-defining plic_set_ie(). You can just make plic_set_ie() public function in PLIC driver.